src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt

```diff
@@ -32,4 +32,10 @@ class ProjectController(private val service: ProjectService) {
         @PathVariable id: Long,
         @RequestBody dto: ProjectDto
     ) = service.updateProjectById(id, dto)
+
+    @PutMapping("/archive/{id}")
+    fun archiveProjectById(@PathVariable id: Long) = service.archiveProjectById(id)
+
+    @PutMapping("/unarchive/{id}")
+    fun unarchiveProjectById(@PathVariable id: Long) = service.unarchiveProjectById(id)
```

Comment:
```suggestion
    @PutMapping("/{id}/archive")
    fun archiveProjectById(@PathVariable id: Long) = service.archiveProjectById(id)

    @PutMapping("/{id}/unarchive")
    fun unarchiveProjectById(@PathVariable id: Long) = service.unarchiveProjectById(id)
```
This is a bit picky but the operation usually comes at the end (it's apparently considered more RESTful). You can look at examples of public APIs:
https://docs.github.com/en/rest/gists/gists#star-a-gist
https://developer.twitter.com/en/docs/twitter-api/tweets/likes/migrate